posts/2024-08-25-five-java-dev-musthaves-fast-startup.adoc

```diff
@@ -0,0 +1,45 @@
+---
+layout: post
+title: "Five Java Developer Must-Haves for Ultra-Fast Startup Solutions"
+# Do NOT change the categories section
+categories: blog
+author_picture: https://avatars3.githubusercontent.com/vijaysun-omr
+author_github: https://github.com/vijaysun-omr
+seo-title: Five Java Developer Must-Haves for Ultra-Fast Startup Solutions - OpenLiberty.io
+seo-description: Faster Java startup should not compromise developer experience, throughput performance, or security. We discuss how we achieved this with Liberty InstantOn.
+blog_description: "Faster Java startup should not compromise developer experience, throughput performance, or security. We discuss how we achieved this with Liberty InstantOn."
+open-graph-image: https://openliberty.io/img/twitter_card.jpg
+open-graph-image-alt: Open Liberty Logo
+additional_authors:
+- name: Thomas Watson
+  github: https://github.com/tjwatson
+  image: https://avatars0.githubusercontent.com/tjwatson
+- name: Laura Cowen
+  github: https://github.com/lauracowen
+  image: https://avatars0.githubusercontent.com/lauracowen
+---
+= Five Java Developer Must-Haves for Ultra-Fast Startup Solutions
+Vijay Sundaresan <https://github.com/vijaysun-omr>
+:imagesdir: /
+:url-prefix:
+:url-about: /
+//Blank line here is necessary before starting the body of the post.
+
+
+As more and more of our applications are being designed and developed to be deployed into the cloud, and we start to move towards microservices and even serverless style application architecture, start-up times are becoming ever more important. Unfortunately, although Java is a fantastic, robust language for us to use for developing enterprise applications, it isn't really known for its quick start-up times. 
+
+However, several solutions have recently emerged to help with this, speeding up Java startup. They range from compiling a link:https://www.graalvm.org/latest/reference-manual/native-image/[native image] that removes the JVM from the compiled app (the startup speed bottleneck in Java apps) to link:https://docs.azul.com/core/crac/crac-introduction[taking a snapshot of the application] after startup so that there are no startup tasks to complete when restored. But, most of these solutions suffer from compromises to developer experience, throughput performance, or security. 
```

Comment:
```suggestion
However, several solutions have recently emerged to help speed up Java startup. They range from compiling a link:https://www.graalvm.org/latest/reference-manual/native-image/[native image] that removes the JVM from the compiled app (the startup speed bottleneck in Java apps) to link:https://docs.azul.com/core/crac/crac-introduction[taking a snapshot of the application] after startup so that there are no startup tasks to complete when it's restored. But, most of these solutions suffer from compromises to developer experience, throughput performance, or security. 
```